modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorFieldMapper.java

```diff
@@ -315,7 +318,7 @@ Query createCandidateQuery(IndexReader indexReader) throws IOException {
         this.queryTermsField = queryTermsField;
         this.extractionResultField = extractionResultField;
         this.queryBuilderField = queryBuilderField;
-        this.mapUnmappedFieldAsString = INDEX_MAP_UNMAPPED_FIELDS_AS_STRING_SETTING.get(indexSettings);
+        this.mapUnmappedFieldAsString = INDEX_MAP_UNMAPPED_FIELDS_AS_TEXT_SETTING.get(indexSettings);
```

Comment:
let's rename the variable too?

Reply:
Also I'm wondering whether we should do the fallback manually instead of just declaring the `_AS_STRING` as a fallback setting. It would allow us to emit a deprecated warning explaining that it has been replaced with the `_AS_TEXT` setting?

Reply:
@jpountz you mean something like this: https://github.com/elastic/elasticsearch/pull/25395/commits/134b1bc91b71264a53832585d99cb12de99b5b6d#diff-7a987e4fb3c8a1820bfca74aa3e8b5c0R330 ?